packages/core/src/types.ts

```diff
@@ -472,26 +472,33 @@ export type TransitionsConfig<
   TAction extends ParameterizedObject,
   TGuard extends ParameterizedObject,
   TDelay extends string
-> = {
-  // TODO: this doesn't support partial descriptors
-  [K in TEvent['type'] | '*']?: K extends '*'
-    ? TransitionConfigOrTarget<
-        TContext,
-        TEvent,
-        TEvent,
-        TAction,
-        TGuard,
-        TDelay
-      >
-    : TransitionConfigOrTarget<
+> =
+  | {
+      [K in EventDescriptor<TEvent>]?: TransitionConfigOrTarget<
```

Comment:
we should still decide if we want to allow actor-related events here (or other auto-generated events like the ones related to timers, `done.state.*` etc)

Reply:
Eventually yes